src/main/java/trackitnus/model/Model.java

```diff
@@ -227,4 +227,6 @@
     void updateFilteredLessonList(Predicate<Lesson> predicate);
 
     Optional<Lesson> getLesson(Code code, Type type);
+
+    void sortLesson();
```

Comment:
Actually for this I wanted to check why you need this method here. Why not just sort the lessons in UniqueLessonList under the function asUnmodfiableList(). Then everytime UI calls for a list of lessons it will be sorted